<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Yunqi Yang" />


<title>Check coxph fit property</title>

<script src="site_libs/header-attrs-2.11/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/master/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>








<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.tab('show');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">survival-susie</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/yunqiyang0215/survival-susie">
    <span class="fa fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Check coxph fit property</h1>
<h4 class="author">Yunqi Yang</h4>
<h4 class="date">6/15/2023</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2023-06-16
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>survival-susie/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.6.2). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date </a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20230201code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20230201)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20230201code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20230201)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomyunqiyang0215survivalsusietree42db011f20b0e634dd41f590febc062919573865targetblank42db011a"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/yunqiyang0215/survival-susie/tree/42db011f20b0e634dd41f590febc062919573865" target="_blank">42db011</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomyunqiyang0215survivalsusietree42db011f20b0e634dd41f590febc062919573865targetblank42db011a" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://github.com/yunqiyang0215/survival-susie/tree/42db011f20b0e634dd41f590febc062919573865" target="_blank">42db011</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    data/.DS_Store

Unstaged changes:
    Modified:   analysis/run_ser_simple_dat.Rmd
    Modified:   analysis/ser_survival.Rmd
    Modified:   data/dsc3/susie.lbf.rds

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were made to the R Markdown (<code>analysis/coxph_fit.Rmd</code>) and HTML (<code>docs/coxph_fit.html</code>) files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/yunqiyang0215/survival-susie/blob/42db011f20b0e634dd41f590febc062919573865/analysis/coxph_fit.Rmd" target="_blank">42db011</a>
</td>
<td>
yunqiyang0215
</td>
<td>
2023-06-16
</td>
<td>
wflow_publish("analysis/coxph_fit.Rmd")
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/yunqiyang0215/survival-susie/2aa5a4cf9cf1e022a27381ffefa5833ed1231818/docs/coxph_fit.html" target="_blank">2aa5a4c</a>
</td>
<td>
yunqiyang0215
</td>
<td>
2023-06-16
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/yunqiyang0215/survival-susie/blob/4b699e2fe059d47aef529683f665bf3722422b61/analysis/coxph_fit.Rmd" target="_blank">4b699e2</a>
</td>
<td>
yunqiyang0215
</td>
<td>
2023-06-16
</td>
<td>
wflow_publish("analysis/coxph_fit.Rmd")
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/yunqiyang0215/survival-susie/f7b9dc74af89443b5966d651a10ada1426c29239/docs/coxph_fit.html" target="_blank">f7b9dc7</a>
</td>
<td>
yunqiyang0215
</td>
<td>
2023-06-16
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/yunqiyang0215/survival-susie/blob/2248e0e742794147581dea923e3ef30e1e9ada85/analysis/coxph_fit.Rmd" target="_blank">2248e0e</a>
</td>
<td>
yunqiyang0215
</td>
<td>
2023-06-16
</td>
<td>
wflow_publish("analysis/coxph_fit.Rmd")
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<p><strong>Conclusion</strong>:</p>
<ol style="list-style-type: decimal">
<li><p>Extremely high censoring rate probably won’t affect effect size estimate as long as there are moderate number of events.</p></li>
<li><p>If we draw a contigency table (<span class="math inline">\(2\times 2\)</span>). <span class="math inline">\(\delta = 0,1\)</span> indicates whether the outcome is observed, and <span class="math inline">\(x=0,1\)</span> indicates genotypes. As long as there are a couple observations in each cell, the effect size estimate is fine. Otherwise, coefficient tends to be infinite.</p></li>
<li><p>What if the SNP is very rare and censoring rate is extremely high?</p></li>
</ol>
<p>As long as the total sample size is large, even there is only 1 event, the fit is not bad.</p>
<pre class="r"><code>library(survival)</code></pre>
<pre class="r"><code># Function to simulate survival time under exponential model. That is,
# assuming survival time is exponentially distributed.
# lambda(t) = lambda*exp(b0 + Xb). S(t) = exp(-lambda*t),
# F(t) = 1- S(t) \sim Unif(0,1). Therefore, t = log(1-runif(0,1))/-exp(b0+Xb).
# For censored objects, we simulate the censoring time by rescale the actual survival time.
# @param b: vector of length (p+1) for true effect size, include intercept.
# @param X: variable matrix of size n by p.
# @param censor_lvl: a constant from [0,1], indicating the censoring level in the data.
# @return  dat: a dataframe that contains `y`, `x` and `status`.
# `status`: censoring status: 0 = censored, 1 = event observed. See Surv() in library(survival)
sim_surv_exp &lt;- function(b, X, censor_lvl){
  n = nrow(X)
  p = ncol(X)
  dat = list()

  status &lt;- ifelse(runif(n) &gt; censor_lvl, 1, 0)
  lambda &lt;- exp(cbind(rep(1,n), X) %*% b)
  surT &lt;- log(1 - runif(n)) /(-lambda)
  # rescale censored subject to get observed time
  surT[status == 0] = surT[status == 0] * runif(sum(status == 0))

  y = cbind(surT, status)
  colnames(y) = c(&quot;time&quot;, &quot;status&quot;)
  colnames(X) &lt;- unlist(lapply(1:p, function(i) paste0(&quot;x&quot;, i)))
  dat[[&quot;X&quot;]] = X
  dat[[&quot;y&quot;]] = y
  return(dat)
}</code></pre>
<div id="check-discriminate-power-of-coxph." class="section level3">
<h3>1. Check discriminate power of coxph().</h3>
<pre class="r"><code>set.seed(1)
# simulate 2 variables 
n = 100
X = cbind(rbinom(n, size = 1, prob = 0.3), rbinom(n, size = 1, prob = 0.1))
# the first element of b is for intercept
b = c(1, 1, 0)
censor_lvl = 0.7
dat &lt;- sim_surv_exp(b, X, censor_lvl)</code></pre>
<div id="fit1-use-all-the-data" class="section level4">
<h4>Fit1: use all the data</h4>
<pre class="r"><code># rownames are unique y[,2] values, delta. delta = 1 indicates event happened. 
# colnames are X values, genotype.
table(dat$y[,2], dat$X[,1])</code></pre>
<pre><code>   
     0  1
  0 53 26
  1 15  6</code></pre>
<pre class="r"><code>surT &lt;- Surv(dat$y[,1], dat$y[,2])
fit1 &lt;- coxph(surT ~ dat$X[,1])
summary(fit1)</code></pre>
<pre><code>Call:
coxph(formula = surT ~ dat$X[, 1])

  n= 100, number of events= 21 

            coef exp(coef) se(coef)     z Pr(&gt;|z|)    
dat$X[, 1] 2.035     7.656    0.587 3.468 0.000525 ***
---
Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1

           exp(coef) exp(-coef) lower .95 upper .95
dat$X[, 1]     7.656     0.1306     2.423     24.19

Concordance= 0.686  (se = 0.058 )
Likelihood ratio test= 10.26  on 1 df,   p=0.001
Wald test            = 12.02  on 1 df,   p=5e-04
Score (logrank) test = 15.51  on 1 df,   p=8e-05</code></pre>
</div>
<div id="fit2" class="section level4">
<h4>Fit2:</h4>
<p>Select sub-samples where individuals who have event have x = 1. And individuals with x = 0 are all censored.</p>
<pre class="r"><code># let&#39;s just select sub-samples where individuals who have event have x = 1. 
# And individuals with x = 0 are all censored. 
indx = which(dat$y[,2] == dat$X[,1])
y = dat$y[indx, ]
x = dat$X[indx, 1]
table(y[,2], x)</code></pre>
<pre><code>   x
     0  1
  0 53  0
  1  0  6</code></pre>
<pre class="r"><code>surT &lt;- Surv(y[,1], y[,2])
fit2 &lt;- coxph(surT ~ x)</code></pre>
<pre><code>Warning in coxph.fit(X, Y, istrat, offset, init, control, weights = weights, :
Loglik converged before variable 1 ; coefficient may be infinite.</code></pre>
<pre class="r"><code>summary(fit2)</code></pre>
<pre><code>Call:
coxph(formula = surT ~ x)

  n= 59, number of events= 6 

       coef exp(coef)  se(coef)     z Pr(&gt;|z|)
x 2.410e+01 2.936e+10 2.077e+04 0.001    0.999

  exp(coef) exp(-coef) lower .95 upper .95
x 2.936e+10  3.406e-11         0       Inf

Concordance= 0.964  (se = 0.015 )
Likelihood ratio test= 29.39  on 1 df,   p=6e-08
Wald test            = 0  on 1 df,   p=1
Score (logrank) test = 60.57  on 1 df,   p=7e-15</code></pre>
</div>
<div id="fit3-fit4" class="section level4">
<h4>Fit3 &amp; Fit4:</h4>
<p>In fit3, we add 2 samples to data in fit2. Doesn’t help in this case.</p>
<p>In fit4, we add 4 samples to data in fit2. It helped a lot.</p>
<pre class="r"><code>indx2 = which(dat$y[,2] != dat$X[,1])[c(1,4)]
y2 = rbind(y, dat$y[indx2, ])
x2 = c(x, dat$X[,1][indx2])
table(y2[,2], x2)</code></pre>
<pre><code>   x2
     0  1
  0 53  1
  1  1  6</code></pre>
<pre class="r"><code>surT &lt;- Surv(y2[,1], y2[,2])
fit3 &lt;- coxph(surT ~ x2)</code></pre>
<pre><code>Warning in coxph.fit(X, Y, istrat, offset, init, control, weights = weights, :
Loglik converged before variable 1 ; coefficient may be infinite.</code></pre>
<pre class="r"><code>summary(fit3)</code></pre>
<pre><code>Call:
coxph(formula = surT ~ x2)

  n= 61, number of events= 7 

        coef exp(coef)  se(coef)     z Pr(&gt;|z|)
x2 2.335e+01 1.379e+10 1.443e+04 0.002    0.999

   exp(coef) exp(-coef) lower .95 upper .95
x2 1.379e+10   7.25e-11         0       Inf

Concordance= 0.919  (se = 0.036 )
Likelihood ratio test= 28.75  on 1 df,   p=8e-08
Wald test            = 0  on 1 df,   p=1
Score (logrank) test = 56.01  on 1 df,   p=7e-14</code></pre>
<pre class="r"><code>indx2 = which(dat$y[,2] != dat$X[,1])[c(1:4)]
y2 = rbind(y, dat$y[indx2, ])
x2 = c(x, dat$X[,1][indx2])
table(y2[,2], x2)</code></pre>
<pre><code>   x2
     0  1
  0 53  3
  1  1  6</code></pre>
<pre class="r"><code>surT &lt;- Surv(y2[,1], y2[,2])
fit4 &lt;- coxph(surT ~ x2)
summary(fit4)</code></pre>
<pre><code>Call:
coxph(formula = surT ~ x2)

  n= 63, number of events= 7 

     coef exp(coef) se(coef)     z Pr(&gt;|z|)    
x2  3.996    54.381    1.099 3.635 0.000278 ***
---
Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1

   exp(coef) exp(-coef) lower .95 upper .95
x2     54.38    0.01839     6.304     469.1

Concordance= 0.904  (se = 0.04 )
Likelihood ratio test= 21.26  on 1 df,   p=4e-06
Wald test            = 13.21  on 1 df,   p=3e-04
Score (logrank) test = 38.8  on 1 df,   p=5e-10</code></pre>
</div>
</div>
<div id="check-coxph-under-extremely-high-censoring." class="section level3">
<h3>2. Check coxph() under extremely high censoring.</h3>
<pre class="r"><code>set.seed(1)
# simulate 2 variables 
n = 1000
X = cbind(rbinom(n, size = 1, prob = 0.3), rbinom(n, size = 1, prob = 0.1))
# the first element of b is for intercept
b = c(1, 1, 0)
censor_lvl = 0.99
dat &lt;- sim_surv_exp(b, X, censor_lvl)</code></pre>
<div id="fit1-use-all-the-data-1" class="section level4">
<h4>Fit1: use all the data</h4>
<p>In this case,</p>
<pre class="r"><code>table(dat$y[,2], dat$X[,1])</code></pre>
<pre><code>   
      0   1
  0 690 301
  1   6   3</code></pre>
<pre class="r"><code>surT &lt;- Surv(dat$y[,1], dat$y[,2])
fit1 &lt;- coxph(surT ~ dat$X[,1])
summary(fit1)</code></pre>
<pre><code>Call:
coxph(formula = surT ~ dat$X[, 1])

  n= 1000, number of events= 9 

             coef exp(coef) se(coef)    z Pr(&gt;|z|)  
dat$X[, 1] 1.4241    4.1541   0.7576 1.88   0.0601 .
---
Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1

           exp(coef) exp(-coef) lower .95 upper .95
dat$X[, 1]     4.154     0.2407    0.9411     18.34

Concordance= 0.699  (se = 0.09 )
Likelihood ratio test= 3.03  on 1 df,   p=0.08
Wald test            = 3.53  on 1 df,   p=0.06
Score (logrank) test = 4.09  on 1 df,   p=0.04</code></pre>
</div>
</div>
<div id="what-if-the-snp-is-very-rare-and-censoring-rate-is-extremely-high." class="section level3">
<h3>3. What if the SNP is very rare and censoring rate is extremely high.</h3>
<p>In this extreme case, the fit is not bad.</p>
<pre class="r"><code>set.seed(1)
# simulate 2 variables 
n = 10000
X = cbind(rbinom(n, size = 2, prob = 5e-3), rbinom(n, size = 1, prob = 0.1))
# the first element of b is for intercept
b = c(1, 1, 0)
censor_lvl = 0.99
dat &lt;- sim_surv_exp(b, X, censor_lvl)</code></pre>
<pre class="r"><code>table(dat$y[,2], dat$X[,1])</code></pre>
<pre><code>   
       0    1
  0 9810   95
  1   94    1</code></pre>
<pre class="r"><code>surT &lt;- Surv(dat$y[,1], dat$y[,2])
fit1 &lt;- coxph(surT ~ dat$X[,1])
summary(fit1)</code></pre>
<pre><code>Call:
coxph(formula = surT ~ dat$X[, 1])

  n= 10000, number of events= 95 

            coef exp(coef) se(coef)     z Pr(&gt;|z|)
dat$X[, 1] 1.380     3.974    1.010 1.366    0.172

           exp(coef) exp(-coef) lower .95 upper .95
dat$X[, 1]     3.974     0.2516    0.5492     28.76

Concordance= 0.511  (se = 0.013 )
Likelihood ratio test= 1.25  on 1 df,   p=0.3
Wald test            = 1.87  on 1 df,   p=0.2
Score (logrank) test = 2.18  on 1 df,   p=0.1</code></pre>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.1.1 (2021-08-10)
Platform: x86_64-apple-darwin20.6.0 (64-bit)
Running under: macOS Monterey 12.0.1

Matrix products: default
BLAS:   /usr/local/Cellar/openblas/0.3.18/lib/libopenblasp-r0.3.18.dylib
LAPACK: /usr/local/Cellar/r/4.1.1_1/lib/R/lib/libRlapack.dylib

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] survival_3.2-11 workflowr_1.6.2

loaded via a namespace (and not attached):
 [1] Rcpp_1.0.8.3     pillar_1.6.4     compiler_4.1.1   bslib_0.4.1     
 [5] later_1.3.0      jquerylib_0.1.4  git2r_0.28.0     tools_4.1.1     
 [9] digest_0.6.28    lattice_0.20-44  jsonlite_1.7.2   evaluate_0.14   
[13] lifecycle_1.0.1  tibble_3.1.5     pkgconfig_2.0.3  rlang_1.0.6     
[17] Matrix_1.5-3     cli_3.1.0        rstudioapi_0.13  yaml_2.2.1      
[21] xfun_0.27        fastmap_1.1.0    stringr_1.4.0    knitr_1.36      
[25] fs_1.5.0         vctrs_0.3.8      sass_0.4.4       grid_4.1.1      
[29] rprojroot_2.0.2  glue_1.4.2       R6_2.5.1         fansi_0.5.0     
[33] rmarkdown_2.11   magrittr_2.0.1   whisker_0.4      splines_4.1.1   
[37] promises_1.2.0.1 ellipsis_0.3.2   htmltools_0.5.5  httpuv_1.6.3    
[41] utf8_1.2.2       stringi_1.7.5    cachem_1.0.6     crayon_1.4.1    </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
