<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Yunqi Yang and Peter Carbonetto" />

<meta name="date" content="2024-12-06" />

<title>Introduction to CoxPH-SuSiE</title>

<script src="site_libs/header-attrs-2.26/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/readable.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-6.4.2/css/all.min.css" rel="stylesheet" />
<link href="site_libs/font-awesome-6.4.2/css/v4-shims.min.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->




</head>

<body>


<div class="container-fluid main-container">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">survival-susie</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/yunqiyang0215/survival-susie">
    <span class="fa fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Introduction to CoxPH-SuSiE</h1>
<h4 class="author">Yunqi Yang and Peter Carbonetto</h4>
<h4 class="date">December 6, 2024</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2024-12-09
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>survival-susie/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.1). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date
</a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git
repository, you know the exact version of the code that produced these
results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20230201code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20230201)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20230201code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20230201)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomyunqiyang0215survivalsusietree92e56711924d56f3ebd81f822e9872719c09a4d3targetblank92e5671a">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/yunqiyang0215/survival-susie/tree/92e56711924d56f3ebd81f822e9872719c09a4d3" target="_blank">92e5671</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomyunqiyang0215survivalsusietree92e56711924d56f3ebd81f822e9872719c09a4d3targetblank92e5671a"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/yunqiyang0215/survival-susie/tree/92e56711924d56f3ebd81f822e9872719c09a4d3" target="_blank">92e5671</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Untracked files:
    Untracked:  analysis/survival_demo.RData

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were
made to the R Markdown (<code>analysis/coxph_susie_intro.Rmd</code>) and
HTML (<code>docs/coxph_susie_intro.html</code>) files. If you’ve
configured a remote Git repository (see <code>?wflow_git_remote</code>),
click on the hyperlinks in the table below to view the files as they
were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/yunqiyang0215/survival-susie/92e56711924d56f3ebd81f822e9872719c09a4d3/docs/coxph_susie_intro.html" target="_blank">92e5671</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2024-12-09
</td>
<td>
Added link to the overview page.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/yunqiyang0215/survival-susie/blob/32e28c5f58228bedd13ea7159dfec4e14cf96348/analysis/coxph_susie_intro.Rmd" target="_blank">32e28c5</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2024-12-09
</td>
<td>
workflowr::wflow_publish("analysis/index.Rmd")
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/yunqiyang0215/survival-susie/d421a8314fd534363eea8ef17f166107f6496bbe/docs/coxph_susie_intro.html" target="_blank">d421a83</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2024-12-06
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/yunqiyang0215/survival-susie/blob/2a72a024110d4ae2efcabd21191cf6b4444d6a87/analysis/coxph_susie_intro.Rmd" target="_blank">2a72a02</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2024-12-06
</td>
<td>
workflowr::wflow_publish("analysis/coxph_susie_intro.Rmd")
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/yunqiyang0215/survival-susie/be527ceca34860941e0c90bc37735d4b5678bce2/docs/coxph_susie_intro.html" target="_blank">be527ce</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2024-12-06
</td>
<td>
First build of the coxph_susie_intro vignette.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/yunqiyang0215/survival-susie/blob/f62c83955069443223290b9262a70d6aaba02216/analysis/coxph_susie_intro.Rmd" target="_blank">f62c839</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2024-12-06
</td>
<td>
workflowr::wflow_publish("analysis/coxph_susie_intro.Rmd")
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<p>This short vignette is intended to introduce the CoxPH-SuSiE model
for fine-mapping time-to-event (TTE) phenotypes, and illustrate how to
use the methods on an example genetic data set.</p>
<p>To run the code below, you will need the survival, susieR and
logisticsusie R packages, which can be installed with these commands in
R:</p>
<pre class="r"><code>install.packages(&quot;remotes&quot;)
install.packages(&quot;survival&quot;)
remotes::install_github(&quot;stephenslab/susieR&quot;)
remotes::install_github(&quot;karltayeb/logisticsusie&quot;)</code></pre>
<p>Also, we will use the ggplot2 and cowplot packages to create some
helpful visualizations of the data and results:</p>
<p><code>{r install-more-packages, eval=FALSE install.packages(c("ggplot2","cowplot"))</code></p>
<p>Once the packages are installed, load them.</p>
<pre class="r"><code>library(survival)
library(susieR)
library(logisticsusie)
library(ggplot2)
library(cowplot)</code></pre>
<p>The example data set is found in the “data” directory of this
repository, or can be downloaded by running these commands in R:</p>
<pre class="r"><code>data_url &lt;- paste0(&quot;https://raw.githubusercontent.com/&quot;,
                   &quot;yunqiyang0215/survival-susie/&quot;,
                   &quot;refs/heads/master/data/survival_demo.RData&quot;)
download.file(data_url,&quot;survival_demo.RData&quot;)
load(&quot;survival_demo.RData&quot;)</code></pre>
<p>In this example, the TTE data are “censored” disease diagnosis times,
meaning that the observations are either: the time of diagnosis for
uncensored samples; or a lower limit on the time of diagnosis for
censored samples, indicated by the “+”. (This is called “right
censoring”.)</p>
<pre class="r"><code>pheno &lt;- Surv(time,status)
head(pheno,n = 10)
#  [1] 0.3085771  0.4654124  2.7448602+ 1.3071323+ 0.2252482  3.3300303+
#  [7] 1.0663338+ 7.4103627+ 0.9918607  0.0637775</code></pre>
<p>Among the 574 diagnosis times in this data set, 330 are observed, and
the remaining 244 are right-censored:</p>
<pre class="r"><code>table(status)
# status
#   0   1 
# 244 330</code></pre>
<p>The “geno” matrix contains the (imputed) genotype data for 1,000
candidate genetic variants (SNPs) positioned near each other on a
chromosome.</p>
<pre class="r"><code>nrow(geno)
ncol(geno)
# [1] 574
# [1] 1000</code></pre>
<p>Basic association tests using the Cox proportional hazards (CoxPH)
model show that many of these genetic variants are associated with the
TTE phenotype:</p>
<pre class="r"><code>pvals &lt;-
  apply(geno,2,
        function (x) summary(coxph(pheno ~ x))$coefficients[1,&quot;Pr(&gt;|z|)&quot;])
pdat &lt;- data.frame(pos = 1:1000,pval = pvals,causal = (b != 0))
ggplot(pdat,aes(x = pos,y = -log10(pval),color = causal)) +
  geom_point(shape = 1,size = 1) +
  scale_color_manual(values = c(&quot;dodgerblue&quot;,&quot;darkorange&quot;)) +
  labs(x = &quot;chromosomal position&quot;,y = &quot;-log10 p-value&quot;) +
  theme_cowplot(font_size = 10)</code></pre>
<p><img src="figure/coxph_susie_intro.Rmd/genome-wide-scan-1.png" width="720" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-genome-wide-scan-1">
Past versions of genome-wide-scan-1.png
</button>
</p>
<div id="fig-genome-wide-scan-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/yunqiyang0215/survival-susie/blob/92e56711924d56f3ebd81f822e9872719c09a4d3/docs/figure/coxph_susie_intro.Rmd/genome-wide-scan-1.png" target="_blank">92e5671</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2024-12-09
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Since we simulated these data, we know that three SNPs truly affect
the TTE phenotype (drawn in orange in the plot). All three of the causal
SNPs show evidence for association from the CoxPH-based association
tests, but the strength of support for association varies quite a bit
among the three SNPs. Further, many more SNPs that do not affect the
phenotype are also associated, and this occurs <em>because many of the
SNPs are strongly correlated with each other</em> (strong <em>linkage
disequilibrium</em>, or “LD” for short).</p>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()
# R version 4.3.3 (2024-02-29)
# Platform: aarch64-apple-darwin20 (64-bit)
# Running under: macOS Sonoma 14.7.1
# 
# Matrix products: default
# BLAS:   /Library/Frameworks/R.framework/Versions/4.3-arm64/Resources/lib/libRblas.0.dylib 
# LAPACK: /Library/Frameworks/R.framework/Versions/4.3-arm64/Resources/lib/libRlapack.dylib;  LAPACK version 3.11.0
# 
# locale:
# [1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8
# 
# time zone: America/Chicago
# tzcode source: internal
# 
# attached base packages:
# [1] stats     graphics  grDevices utils     datasets  methods   base     
# 
# other attached packages:
# [1] cowplot_1.1.3            ggplot2_3.5.0            logisticsusie_0.0.0.9002
# [4] susieR_0.12.42           survival_3.5-8           workflowr_1.7.1         
# 
# loaded via a namespace (and not attached):
#  [1] generics_0.1.3    sass_0.4.8        utf8_1.2.4        stringi_1.8.3    
#  [5] lattice_0.22-5    digest_0.6.34     magrittr_2.0.3    evaluate_0.23    
#  [9] grid_4.3.3        fastmap_1.1.1     plyr_1.8.9        rprojroot_2.0.4  
# [13] jsonlite_1.8.8    Matrix_1.6-5      processx_3.8.3    whisker_0.4.1    
# [17] reshape_0.8.9     mixsqp_0.3-54     ps_1.7.6          promises_1.2.1   
# [21] httr_1.4.7        fansi_1.0.6       scales_1.3.0      jquerylib_0.1.4  
# [25] cli_3.6.2         rlang_1.1.3       crayon_1.5.2      munsell_0.5.0    
# [29] splines_4.3.3     withr_3.0.0       cachem_1.0.8      yaml_2.3.8       
# [33] tools_4.3.3       dplyr_1.1.4       colorspace_2.1-0  httpuv_1.6.14    
# [37] vctrs_0.6.5       R6_2.5.1          matrixStats_1.2.0 lifecycle_1.0.4  
# [41] git2r_0.33.0      stringr_1.5.1     fs_1.6.3          irlba_2.3.5.1    
# [45] pkgconfig_2.0.3   callr_3.7.5       pillar_1.9.0      bslib_0.6.1      
# [49] later_1.3.2       gtable_0.3.4      glue_1.7.0        Rcpp_1.0.12      
# [53] highr_0.10        tidyselect_1.2.1  xfun_0.42         tibble_3.2.1     
# [57] rstudioapi_0.15.0 knitr_1.45        farver_2.1.1      htmltools_0.5.7  
# [61] labeling_0.4.3    rmarkdown_2.26    compiler_4.3.3    getPass_0.2-4</code></pre>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>





</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
