<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Yunqi Yang, Karl Tayeb, Peter Carbonetto and Matthew Stephens" />

<meta name="date" content="2024-12-10" />

<title>Introduction to CoxPH-SuSiE</title>

<script src="site_libs/header-attrs-2.26/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/readable.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-6.4.2/css/all.min.css" rel="stylesheet" />
<link href="site_libs/font-awesome-6.4.2/css/v4-shims.min.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->




</head>

<body>


<div class="container-fluid main-container">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">survival-susie</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/yunqiyang0215/survival-susie">
    <span class="fa fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Introduction to CoxPH-SuSiE</h1>
<h4 class="author">Yunqi Yang, Karl Tayeb, Peter Carbonetto and Matthew
Stephens</h4>
<h4 class="date">December 10, 2024</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2024-12-18
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>survival-susie/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.1). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date
</a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git
repository, you know the exact version of the code that produced these
results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20230201code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20230201)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20230201code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20230201)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomyunqiyang0215survivalsusietreee6ba60e0e90baefcfcc819ead3ec3fbdffd31e6dtargetblanke6ba60ea">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/yunqiyang0215/survival-susie/tree/e6ba60e0e90baefcfcc819ead3ec3fbdffd31e6d" target="_blank">e6ba60e</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomyunqiyang0215survivalsusietreee6ba60e0e90baefcfcc819ead3ec3fbdffd31e6dtargetblanke6ba60ea"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/yunqiyang0215/survival-susie/tree/e6ba60e0e90baefcfcc819ead3ec3fbdffd31e6d" target="_blank">e6ba60e</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    analysis/.DS_Store

Untracked files:
    Untracked:  survival_demo.RData

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were
made to the R Markdown (<code>analysis/coxph_susie_intro.Rmd</code>) and
HTML (<code>docs/coxph_susie_intro.html</code>) files. If you’ve
configured a remote Git repository (see <code>?wflow_git_remote</code>),
click on the hyperlinks in the table below to view the files as they
were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/yunqiyang0215/survival-susie/blob/e6ba60e0e90baefcfcc819ead3ec3fbdffd31e6d/analysis/coxph_susie_intro.Rmd" target="_blank">e6ba60e</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2024-12-18
</td>
<td>
workflowr::wflow_publish("analysis/coxph_susie_intro.Rmd", verbose =
TRUE,
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/yunqiyang0215/survival-susie/ede61684f9cdb3b6db8a4feadad8891a76c9c1a9/docs/coxph_susie_intro.html" target="_blank">ede6168</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2024-12-18
</td>
<td>
Added prior variance estimation and the SER BFs to the
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/yunqiyang0215/survival-susie/blob/078933910a23328c3bf7d361adfda1ed08c154aa/analysis/coxph_susie_intro.Rmd" target="_blank">0789339</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2024-12-18
</td>
<td>
workflowr::wflow_publish("analysis/coxph_susie_intro.Rmd", verbose =
TRUE,
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/yunqiyang0215/survival-susie/9189825b2ed4dceddd2c221bdbb6ab4ec9960fb8/docs/coxph_susie_intro.html" target="_blank">9189825</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2024-12-13
</td>
<td>
In coxph_susie_intro demo, added call to ibss_from_ser() with
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/yunqiyang0215/survival-susie/blob/e483e17ca9888009164247a1e4d47192e6a3312c/analysis/coxph_susie_intro.Rmd" target="_blank">e483e17</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2024-12-13
</td>
<td>
workflowr::wflow_publish("analysis/coxph_susie_intro.Rmd", verbose =
TRUE)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/yunqiyang0215/survival-susie/dd1fd56805b8416289da4f80376db1d672da3263/docs/coxph_susie_intro.html" target="_blank">dd1fd56</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2024-12-10
</td>
<td>
Updated the coxph_susie_intro page.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/yunqiyang0215/survival-susie/blob/0dec6498f7bc7194c3cdfbf3f73083516fa7b039/analysis/coxph_susie_intro.Rmd" target="_blank">0dec649</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2024-12-10
</td>
<td>
Updated authors and date for coxph_susie_intro vignette.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/yunqiyang0215/survival-susie/183f1fba33063fe4eba9bb19e8e6afac7f1803a3/docs/coxph_susie_intro.html" target="_blank">183f1fb</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2024-12-10
</td>
<td>
Added comparison of coefs to coxph_susie_intro demo.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/yunqiyang0215/survival-susie/blob/991bd75a7678cee713ef16be46f71fd9f7d6bcd6/analysis/coxph_susie_intro.Rmd" target="_blank">991bd75</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2024-12-10
</td>
<td>
workflowr::wflow_publish("coxph_susie_intro.Rmd", verbose = TRUE,
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/yunqiyang0215/survival-susie/de9be109f136d445d354c9806736c37fa95f3105/docs/coxph_susie_intro.html" target="_blank">de9be10</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2024-12-09
</td>
<td>
Small fix to one of the plots in the introductory vignette.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/yunqiyang0215/survival-susie/blob/35b02892b74093b433d6de33f2027092302c483d/analysis/coxph_susie_intro.Rmd" target="_blank">35b0289</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2024-12-09
</td>
<td>
workflowr::wflow_publish("coxph_susie_intro.Rmd", verbose = TRUE)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/yunqiyang0215/survival-susie/7442aeb71fa0f3cee525f9489ffb79d9a518912f/docs/coxph_susie_intro.html" target="_blank">7442aeb</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2024-12-09
</td>
<td>
Added more explanatory text to the introductory vignette.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/yunqiyang0215/survival-susie/blob/10c35d03d7af11f54d31279b53ea111cccf936cb/analysis/coxph_susie_intro.Rmd" target="_blank">10c35d0</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2024-12-09
</td>
<td>
workflowr::wflow_publish("coxph_susie_intro.Rmd", verbose = TRUE)
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/yunqiyang0215/survival-susie/blob/caed3b02ea52c28cf6d8c7b59fa3f29b1982b19f/analysis/coxph_susie_intro.Rmd" target="_blank">caed3b0</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2024-12-09
</td>
<td>
Made a few improvements to the text of the intro vignette.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/yunqiyang0215/survival-susie/blob/b323562b3c9297c9ae4c599d7300b698c133307d/analysis/coxph_susie_intro.Rmd" target="_blank">b323562</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2024-12-09
</td>
<td>
Added fine-mapping step and fine-mapping plots to coxph_susie_intro Rmd.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/yunqiyang0215/survival-susie/blob/03283885954d03256b2b056e2cae79a0ada54072/analysis/coxph_susie_intro.Rmd" target="_blank">0328388</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2024-12-09
</td>
<td>
Added a bit of text to the coxph_susie_intro vignette.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/yunqiyang0215/survival-susie/b849946358141cb2bb0c576dff8a8f54d12ce7c3/docs/coxph_susie_intro.html" target="_blank">b849946</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2024-12-09
</td>
<td>
Added some material to the coxph-susie demo.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/yunqiyang0215/survival-susie/92e56711924d56f3ebd81f822e9872719c09a4d3/docs/coxph_susie_intro.html" target="_blank">92e5671</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2024-12-09
</td>
<td>
Added link to the overview page.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/yunqiyang0215/survival-susie/blob/32e28c5f58228bedd13ea7159dfec4e14cf96348/analysis/coxph_susie_intro.Rmd" target="_blank">32e28c5</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2024-12-09
</td>
<td>
workflowr::wflow_publish("analysis/index.Rmd")
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/yunqiyang0215/survival-susie/d421a8314fd534363eea8ef17f166107f6496bbe/docs/coxph_susie_intro.html" target="_blank">d421a83</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2024-12-06
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/yunqiyang0215/survival-susie/blob/2a72a024110d4ae2efcabd21191cf6b4444d6a87/analysis/coxph_susie_intro.Rmd" target="_blank">2a72a02</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2024-12-06
</td>
<td>
workflowr::wflow_publish("analysis/coxph_susie_intro.Rmd")
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/yunqiyang0215/survival-susie/be527ceca34860941e0c90bc37735d4b5678bce2/docs/coxph_susie_intro.html" target="_blank">be527ce</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2024-12-06
</td>
<td>
First build of the coxph_susie_intro vignette.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/yunqiyang0215/survival-susie/blob/f62c83955069443223290b9262a70d6aaba02216/analysis/coxph_susie_intro.Rmd" target="_blank">f62c839</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2024-12-06
</td>
<td>
workflowr::wflow_publish("analysis/coxph_susie_intro.Rmd")
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<p>This short vignette is intended to (i) introduce CoxPH-SuSiE for
fine-mapping time-to-event (TTE) phenotypes, and (ii) illustrate how to
use CoxPH-SuSiE on an example genetic data set.</p>
<div id="initial-setup" class="section level2">
<h2>Initial setup</h2>
<p>To run the code below, you will need the survival, susieR and
logisticsusie R packages, which can be installed with these commands in
R:</p>
<pre class="r"><code>install.packages(&quot;remotes&quot;)
install.packages(&quot;survival&quot;)
remotes::install_github(&quot;stephenslab/susieR&quot;)
remotes::install_github(&quot;karltayeb/logisticsusie&quot;)</code></pre>
<p>Also, we will use the ggplot2 and cowplot packages to create some
helpful visualizations of the data and results:</p>
<pre class="r"><code>install.packages(&quot;ggplot2&quot;)
install.packages(&quot;cowplot&quot;)</code></pre>
<p>Once the packages are installed, load them.</p>
<pre class="r"><code>library(survival)
library(susieR)
library(logisticsusie)
library(ggplot2)
library(cowplot)</code></pre>
<p>Set the seed for reproducibility.</p>
<pre class="r"><code>set.seed(1)</code></pre>
</div>
<div id="the-data-set" class="section level2">
<h2>The data set</h2>
<p>The example data set is found in the “data” directory of this GitHub
repository, or can be downloaded by running these commands in R:</p>
<pre class="r"><code>data_url &lt;- paste0(&quot;https://raw.githubusercontent.com/&quot;,
                   &quot;yunqiyang0215/survival-susie/&quot;,
                   &quot;refs/heads/master/data/survival_demo.RData&quot;)
download.file(data_url,&quot;survival_demo.RData&quot;)
load(&quot;survival_demo.RData&quot;)</code></pre>
<p>In this example, the TTE data are “censored” disease diagnosis times,
meaning that the observations are either: the time of diagnosis (for
uncensored samples); or a lower limit on the time of diagnosis for
censored samples. The censored samples are indicated by the “+” (this is
called “right censoring”):</p>
<pre class="r"><code>pheno &lt;- Surv(time,status)
head(pheno,n = 10)
#  [1] 0.3085771  0.4654124  2.7448602+ 1.3071323+ 0.2252482  3.3300303+
#  [7] 1.0663338+ 7.4103627+ 0.9918607  0.0637775</code></pre>
<p>Among the 574 diagnosis times in this data set, 330 are observed, and
the remaining 244 are right-censored:</p>
<pre class="r"><code>table(status)
# status
#   0   1 
# 244 330</code></pre>
<p>The “geno” matrix contains the (imputed) genotype data for 1,000
candidate genetic variants (SNPs) located near each other on the
chromosome.</p>
<pre class="r"><code>nrow(geno)
ncol(geno)
# [1] 574
# [1] 1000</code></pre>
</div>
<div id="association-analysis-using-the-coxph-model"
class="section level2">
<h2>Association analysis using the CoxPH model</h2>
<p>Association tests performed using the Cox proportional hazards
(CoxPH) model show that many of thee genetic variants are associated
with diagnosis time:</p>
<pre class="r"><code>pvals &lt;-
  apply(geno,2,
        function (x) summary(coxph(pheno ~ x))$coefficients[1,&quot;Pr(&gt;|z|)&quot;])
pdat &lt;- data.frame(pos = 1:1000,pval = pvals,causal = (b != 0))
ggplot(pdat,aes(x = pos,y = -log10(pval),color = causal,shape = causal)) +
  geom_point(size = 1.25) +
  scale_color_manual(values = c(&quot;black&quot;,&quot;darkorange&quot;)) +
  scale_shape_manual(values = c(1,17)) +
  labs(x = &quot;chromosomal position&quot;,y = &quot;-log10 p-value&quot;) +
  theme_cowplot(font_size = 10)</code></pre>
<p><img src="figure/coxph_susie_intro.Rmd/genome-wide-scan-1.png" width="720" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-genome-wide-scan-1">
Past versions of genome-wide-scan-1.png
</button>
</p>
<div id="fig-genome-wide-scan-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/yunqiyang0215/survival-susie/blob/de9be109f136d445d354c9806736c37fa95f3105/docs/figure/coxph_susie_intro.Rmd/genome-wide-scan-1.png" target="_blank">de9be10</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2024-12-09
</td>
</tr>
<tr>
<td>
<a href="https://github.com/yunqiyang0215/survival-susie/blob/7442aeb71fa0f3cee525f9489ffb79d9a518912f/docs/figure/coxph_susie_intro.Rmd/genome-wide-scan-1.png" target="_blank">7442aeb</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2024-12-09
</td>
</tr>
<tr>
<td>
<a href="https://github.com/yunqiyang0215/survival-susie/blob/92e56711924d56f3ebd81f822e9872719c09a4d3/docs/figure/coxph_susie_intro.Rmd/genome-wide-scan-1.png" target="_blank">92e5671</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2024-12-09
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Since we simulated these data, we know that 3 of the 1,000 SNPs truly
affect the TTE phenotype (drawn in orange in the plot above). All three
of these “causal” SNPs show evidence for association from the
CoxPH-based association tests, but the strength of support for
association varies quite a bit among the three SNPs. Further, many more
SNPs that do not affect the phenotype are also strongly associated with
the phenotype. This happens <em>because many of the SNPs are strongly
correlated with each other</em> (in strong <em>linkage
disequilibrium</em>, or “LD”).</p>
<p>Next we will run the CoxPH-SuSiE fine-mapping method to try and
narrow the association signal down to a smaller set of causal
candidates.</p>
</div>
<div id="fine-mapping-analysis-using-the-coxph-model"
class="section level2">
<h2>Fine-mapping analysis using the CoxPH model</h2>
<p>To run the fine-mapping, we first need to define a function that
performs a (Bayesian) univariate association analysis (here, using the
CoxPH model). More precisely, this function takes as input the
regression data (X, Y, as well as an additional “offset” term), and the
prior variance of the regression coefficient (v0), and outputs four
quantities: (1) the posterior mean of the regression coefficient; (2)
its posterior variance; (3) the log-Bayes factor comparing the model
against the “null” model (when the coefficient is zero); and (4) an
estimate of the prior variance parameter (here we use the
maximum-likelihood estimate, which is simply the posterior second moment
of the regression coefficient. (This last quantity can be a constant if
you want to keep the prior variance at a fixed value.)</p>
<pre class="r"><code>surv_uni_fun &lt;- function (x, y, e, prior_variance,
                          estimate_intercept = 0, ...) {
  v0   &lt;- prior_variance                         
  fit  &lt;- coxph(y ~ x + offset(e))
  out  &lt;- summary(fit)$coefficients
  bhat &lt;- out[1,&quot;coef&quot;]
  s    &lt;- out[1,&quot;se(coef)&quot;]
  z    &lt;- bhat/s
  lbf  &lt;- log(s^2/(v0 + s^2))/2 + z^2/2*v0/(v0 + s^2)
  lbf  &lt;- lbf - z^2/2 + summary(fit)$logtest[&quot;test&quot;]/2
  v1   &lt;- 1/(1/v0 + 1/s^2)
  mu1  &lt;- v1*bhat/s^2
  return(list(mu = mu1,var = v1,lbf = lbf,
              prior_variance = mu1^2 + v1,
              intercept = 0))
}</code></pre>
<p>Now we can fit the CoxPH-based SuSiE model using the
<code>ibss_from_ser()</code> function from the logisticsusie package.
Note that this step may take a minute or two to run depending on the
computing platform:</p>
<pre class="r"><code>t0 &lt;- proc.time()
fit &lt;- ibss_from_ser(geno,pheno,L = 3,tol = 0.0001,maxit = 100,
                     ser_function = ser_from_univariate(surv_uni_fun))
t1 &lt;- proc.time()
print(t1 - t0)
# 7.216 sec elapsed
#    user  system elapsed 
#   7.052   0.169   7.222</code></pre>
<p>(As an aside, note that the univariate regression computations are
trivially parallelizable, so you have a multicore computing platform,
you can make the method run faster, e.g.,</p>
<pre class="r"><code>t0 &lt;- proc.time()
fit &lt;- ibss_from_ser(geno,pheno,L = 3,tol = 0.0001,maxit = 100,
                     ser_function = ser_from_univariate(surv_uni_fun),
                     num_cores = 4)
t1 &lt;- proc.time()
print(t1 - t0)
# 2.586 sec elapsed
#    user  system elapsed 
#   8.151   1.504   2.590</code></pre>
<p>In this run we were able to achieve about a 3x speedup with 4
threads.)</p>
<p>Now we use the <code>susie_get_cs()</code> function from the susieR
package to compute credible sets (CSs):</p>
<pre class="r"><code>class(fit) &lt;- c(&quot;susie&quot;,&quot;list&quot;)
out &lt;- susie_get_cs(fit,geno,coverage = 0.95,min_abs_cor = 0.1)
out$cs
# $L2
# [1] 994
# 
# $L1
# [1] 550 555 561 566 582 598</code></pre>
<p>Interestingly, in one credible set, CoxPH-SuSiE is able to identify a
single causal SNP with very high confidence (shown in blue in the plot).
Additionally, a second set of causal candidates is identified with 6
plausible candidates (shown in orange). <em>Both of these CSs contain
the true causal SNPs.</em></p>
<pre class="r"><code>pdat &lt;- cbind(pdat,CS = 0)
pdat[out$cs$L1,&quot;CS&quot;] &lt;- 1
pdat[out$cs$L2,&quot;CS&quot;] &lt;- 2
pdat &lt;- transform(pdat,CS = factor(CS))
cs_colors &lt;- c(&quot;black&quot;,&quot;darkorange&quot;,&quot;dodgerblue&quot;)
ggplot(pdat,aes(x = pos,y = -log10(pval),color = CS,shape = causal)) +
  geom_point(size = 1.25) +
  scale_color_manual(values = cs_colors) +
  scale_shape_manual(values = c(1,17)) +
  labs(x = &quot;chromosomal position&quot;,y = &quot;-log10 p-value&quot;) +
  theme_cowplot(font_size = 10)</code></pre>
<p><img src="figure/coxph_susie_intro.Rmd/plot-credible-sets-1.png" width="720" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-plot-credible-sets-1">
Past versions of plot-credible-sets-1.png
</button>
</p>
<div id="fig-plot-credible-sets-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/yunqiyang0215/survival-susie/blob/7442aeb71fa0f3cee525f9489ffb79d9a518912f/docs/figure/coxph_susie_intro.Rmd/plot-credible-sets-1.png" target="_blank">7442aeb</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2024-12-09
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>The PIPs (“posterior inclusion probabilities”) measure the relative
support for the SNPs within each CS. Plotting the PIPs for all SNPs is
helpful for understanding precisely which SNPs are the top
candidates:</p>
<pre class="r"><code>pdat &lt;- cbind(pdat,pip = fit$pip)
ggplot(pdat,aes(x = pos,y = pip,color = CS,shape = causal)) +
  geom_point(size = 1.25) +
  scale_color_manual(values = cs_colors) +
  scale_shape_manual(values = c(1,17)) +
  labs(x = &quot;chromosomal position&quot;,y = &quot;PIP&quot;) +
  theme_cowplot(font_size = 10)</code></pre>
<p><img src="figure/coxph_susie_intro.Rmd/pip-plot-1.png" width="720" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-pip-plot-1">
Past versions of pip-plot-1.png
</button>
</p>
<div id="fig-pip-plot-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/yunqiyang0215/survival-susie/blob/9189825b2ed4dceddd2c221bdbb6ab4ec9960fb8/docs/figure/coxph_susie_intro.Rmd/pip-plot-1.png" target="_blank">9189825</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2024-12-13
</td>
</tr>
<tr>
<td>
<a href="https://github.com/yunqiyang0215/survival-susie/blob/7442aeb71fa0f3cee525f9489ffb79d9a518912f/docs/figure/coxph_susie_intro.Rmd/pip-plot-1.png" target="_blank">7442aeb</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2024-12-09
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>For example, in CS 1 (orange), 3 out of the 6 SNPs are much stronger
candidates than the others.</p>
<p>It is also interesting that the third causal SNPs that was not
identified does show some modest evidence for being a causal SNP in the
PIP plot.</p>
<p>Let’s compare the coefficient estimates against the ground-truth
estimates. These are contained in the “mu” output. Here we see that the
first CS (L1) accurately estimates the coefficient for the second causal
SNP (the true value is -2), and the second CS (L2) accurately estimates
the coefficient for the third causal SNP (the true value is 2).</p>
<pre class="r"><code>rownames(fit$mu) &lt;- c(&quot;L1&quot;,&quot;L2&quot;,&quot;L3&quot;)
causal_snps &lt;- which(b != 0)
round(rbind(b,fit$mu)[,causal_snps],digits = 3)
#     [,1]   [,2]  [,3]
# b  1.000 -2.000 2.000
# L1 1.215 -2.088 0.252
# L2 0.834 -0.076 1.927
# L3 0.988 -0.063 0.127</code></pre>
<p>Finally, note that CoxPH-SuSiE is robust to overestimating the number
of causal SNPs. For example, suppose we set the number of causal SNPs to
10:</p>
<pre class="r"><code>fit &lt;- ibss_from_ser(geno,pheno,L = 10,tol = 0.0001,maxit = 100,
                     ser_function = ser_from_univariate(surv_uni_fun))</code></pre>
<p>The log-Bayes factors quantifying support for each CS (i.e., each
causal SNP) are in the “lbf_ser” output:</p>
<pre class="r"><code>fit$lbf_ser
#         L1         L2         L3         L4         L5         L6         L7 
# 113.309194   8.742860  -1.467698  -1.465683  -1.463014  -1.460718  -1.459312 
#         L8         L9        L10 
#  -1.458954  -1.459594  -1.461082</code></pre>
<p>After the first two CSs, the remaining 8 CSs provide very little
improvement in fit. This result tells us that we can focus on the
results for the first two CSs and ignore the rest. (Note these log-Bayes
factors are computed using the natural logarithm.)</p>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()
# R version 4.3.3 (2024-02-29)
# Platform: aarch64-apple-darwin20 (64-bit)
# Running under: macOS Sonoma 14.7.1
# 
# Matrix products: default
# BLAS:   /Library/Frameworks/R.framework/Versions/4.3-arm64/Resources/lib/libRblas.0.dylib 
# LAPACK: /Library/Frameworks/R.framework/Versions/4.3-arm64/Resources/lib/libRlapack.dylib;  LAPACK version 3.11.0
# 
# locale:
# [1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8
# 
# time zone: America/Chicago
# tzcode source: internal
# 
# attached base packages:
# [1] stats     graphics  grDevices utils     datasets  methods   base     
# 
# other attached packages:
# [1] cowplot_1.1.3            ggplot2_3.5.0            logisticsusie_0.0.0.9007
# [4] susieR_0.12.42           survival_3.5-8           workflowr_1.7.1         
# 
# loaded via a namespace (and not attached):
#  [1] gtable_0.3.4       xfun_0.42          bslib_0.6.1        processx_3.8.3    
#  [5] lattice_0.22-5     callr_3.7.5        vctrs_0.6.5        tools_4.3.3       
#  [9] ps_1.7.6           generics_0.1.3     parallel_4.3.3     tibble_3.2.1      
# [13] fansi_1.0.6        highr_0.10         pkgconfig_2.0.3    Matrix_1.6-5      
# [17] RcppParallel_5.1.7 lifecycle_1.0.4    compiler_4.3.3     farver_2.1.1      
# [21] stringr_1.5.1      git2r_0.33.0       tictoc_1.2.1       munsell_0.5.0     
# [25] getPass_0.2-4      httpuv_1.6.14      htmltools_0.5.7    sass_0.4.8        
# [29] yaml_2.3.8         later_1.3.2        pillar_1.9.0       crayon_1.5.2      
# [33] jquerylib_0.1.4    whisker_0.4.1      cachem_1.0.8       tidyselect_1.2.1  
# [37] digest_0.6.34      stringi_1.8.3      purrr_1.0.2        dplyr_1.1.4       
# [41] labeling_0.4.3     splines_4.3.3      rprojroot_2.0.4    fastmap_1.1.1     
# [45] grid_4.3.3         colorspace_2.1-0   cli_3.6.2          magrittr_2.0.3    
# [49] Rfast_2.1.0        utf8_1.2.4         withr_3.0.0        scales_1.3.0      
# [53] promises_1.2.1     RcppZiggurat_0.1.6 rmarkdown_2.26     httr_1.4.7        
# [57] matrixStats_1.2.0  evaluate_0.23      knitr_1.45         irlba_2.3.5.1     
# [61] rlang_1.1.3        Rcpp_1.0.12        mixsqp_0.3-54      glue_1.7.0        
# [65] rstudioapi_0.15.0  reshape_0.8.9      jsonlite_1.8.8     R6_2.5.1          
# [69] plyr_1.8.9         fs_1.6.3</code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>





</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
