---
title: "Compare PIPs: small sample with default iterations"
author: "Yunqi Yang"
date: "4/6/2023"
output: html_document
---

### Description:

Simulation results based on real genotype data from GTEx. I took the SNPs for a certain gene, Thyroid.ENSG00000132855. Therefore, sample size $n=584$, and I choose variable number $p=1000$.

#### Steps for simulating survival data:

1. Select a random point on the genome, indx_start. Then the predictors are from [indx_start:(indx_start + p -1)]. 

2. Two correlation types: real correlation and independent. In independent, I simply permute each variable values. The max correlation can be 0.3, but a lot of correlation will be close to 0. 

4. Effect size $b\sim N(0,1)$.

5. Simulate survival time using exponential survival model. 


#### Fitting:

All the methods were set to default number of iterations. 

1. susie & susie rss: $L=5$ and estimate prior covariance. niter = 100. 

2. bvsnlp: niter = 30.

3. survival.svb: niter = 1000

4. r2b: 1e6


```{r knitr, echo=FALSE}
knitr::opts_chunk$set(comment = "#",results = "hold",collapse = TRUE,
                      fig.align = "center", warning = FALSE)
```

```{r}
source("/project2/mstephens/yunqiyang/surv-susie/survival-susie/code/pip.R")
```

```{r}
susie = readRDS("/project2/mstephens/yunqiyang/surv-susie/sim2024/sim_default_iter/susie.rds")
svb = readRDS("/project2/mstephens/yunqiyang/surv-susie/sim2024/sim_default_iter/svb.rds")
bvsnlp = readRDS("/project2/mstephens/yunqiyang/surv-susie/sim2024/sim_default_iter/bvsnlp.rds")
r2b = readRDS("/project2/mstephens/yunqiyang/surv-susie/sim2024/sim_default_iter/r2b.rds")
rss = readRDS("/project2/mstephens/yunqiyang/surv-susie/sim2024/sim_default_iter/rss.rds")
```

```{r}
censor_lvls = unique(susie$simulate.censor_lvl)
```

### 1. Use the real correlation of SNPs
```{r fig.width= 12, fig.height=18}
par(mfrow = c(5,4))
for (i in 1:length(censor_lvls)){
  indx = which(susie$simulate.cor_type == "real" & susie$simulate.censor_lvl == censor_lvls[i])
  pip.susie = unlist(lapply(indx, function(x) susie$susie.pip[[x]]))
  pip.svb = unlist(lapply(indx, function(x) svb$svb.pip[[x]]))
  pip.bvsnlp = unlist(lapply(indx, function(x) bvsnlp$bvsnlp.pip[[x]]))
  pip.r2b = unlist(lapply(indx, function(x) r2b$r2b.pip[[x]]))
  pip.rss = unlist(lapply(indx, function(x) rss$rss.pip[[x]]))
  
  is_effect = unlist(lapply(indx, function(x) susie$simulate.is_effect[[x]]))

  res.pip = compare_pip(pip.susie, pip.svb, is_effect)
  plot_pip(res.pip, labs = c("susie", "survival.svb"), main = paste0("censor=",censor_lvls[i]))
  abline(a = 0, b = 1, lty = 2, col = "blue")
  
  res.pip = compare_pip(pip.susie, pip.bvsnlp, is_effect)
  plot_pip(res.pip, labs = c("susie", "bvsnlp"), main = paste0("censor=",censor_lvls[i]))
  abline(a = 0, b = 1, lty = 2, col = "blue")
  
  res.pip = compare_pip(pip.susie, pip.r2b, is_effect)
  plot_pip(res.pip, labs = c("susie", "r2b"), main = paste0("censor=",censor_lvls[i]))
  abline(a = 0, b = 1, lty = 2, col = "blue")
  
  res.pip = compare_pip(pip.susie, pip.rss, is_effect)
  plot_pip(res.pip, labs = c("susie", "rss"), main = paste0("censor=",censor_lvls[i]))
  abline(a = 0, b = 1, lty = 2, col = "blue")
  
}
```


```{r fig.width= 12, fig.height=18}
par(mfrow = c(5,5))
for (i in 1:length(censor_lvls)){
  indx = which(bvsnlp$simulate.cor_type == "real" & bvsnlp$simulate.censor_lvl == censor_lvls[i])
  is_effect = unlist(lapply(indx, function(x) bvsnlp$simulate.is_effect[[x]]))
  pip.susie = unlist(lapply(indx, function(x) susie$susie.pip[[x]]))
  pip.survsvb = unlist(lapply(indx, function(x) svb$svb.pip[[x]]))
  pip.bvsnlp = unlist(lapply(indx, function(x) bvsnlp$bvsnlp.pip[[x]]))
  pip.rss = unlist(lapply(indx, function(x) rss$rss.pip[[x]]))
  pip.r2b = unlist(lapply(indx, function(x) r2b$r2b.pip[[x]]))
  
  # Calibration plot for pip.susie
  calibration.susie = pip_calibration(pip.susie, is_effect)
  plot_calibration(calibration.susie, main = paste0("susie: censor=", censor_lvls[i]))
  
  # Calibration plot for pip.svb
  calibration.survsvb = pip_calibration(pip.survsvb, is_effect)
  plot_calibration(calibration.survsvb, main = paste0("svb: censor=", censor_lvls[i]))
  
  # Calibration plot for pip.bvsnlp
  calibration.bvsnlp = pip_calibration(pip.bvsnlp, is_effect)
  plot_calibration(calibration.bvsnlp, main = paste0("bvsnlp: censor=", censor_lvls[i]))
  
  # Calibration plot for pip.rss
  calibration.rss = pip_calibration(pip.rss, is_effect)
  plot_calibration(calibration.rss, main = paste0("rss: censor=", censor_lvls[i]))
  
  # Calibration plot for pip.r2b
  calibration.r2b = pip_calibration(pip.r2b, is_effect)
  plot_calibration(calibration.r2b, main = paste0("r2b: censor=", censor_lvls[i]))
}
```

### 2. Near independent SNPs

```{r fig.width= 12, fig.height=18}
par(mfrow = c(5,4))
for (i in 1:length(censor_lvls)){
  indx = which(susie$simulate.cor_type == "independent" & susie$simulate.censor_lvl == censor_lvls[i])
  pip.susie = unlist(lapply(indx, function(x) susie$susie.pip[[x]]))
  pip.svb = unlist(lapply(indx, function(x) svb$svb.pip[[x]]))
  pip.bvsnlp = unlist(lapply(indx, function(x) bvsnlp$bvsnlp.pip[[x]]))
  pip.r2b = unlist(lapply(indx, function(x) r2b$r2b.pip[[x]]))
  pip.rss = unlist(lapply(indx, function(x) rss$rss.pip[[x]]))
  
  is_effect = unlist(lapply(indx, function(x) susie$simulate.is_effect[[x]]))


  res.pip = compare_pip(pip.susie, pip.svb, is_effect)
  plot_pip(res.pip, labs = c("susie", "survival.svb"), main = paste0("censor=",censor_lvls[i]))
  abline(a = 0, b = 1, lty = 2, col = "blue")
  
  res.pip = compare_pip(pip.susie, pip.bvsnlp, is_effect)
  plot_pip(res.pip, labs = c("susie", "bvsnlp"), main = paste0("censor=",censor_lvls[i]))
  abline(a = 0, b = 1, lty = 2, col = "blue")
  
  res.pip = compare_pip(pip.susie, pip.r2b, is_effect)
  plot_pip(res.pip, labs = c("susie", "r2b"), main = paste0("censor=",censor_lvls[i]))
  abline(a = 0, b = 1, lty = 2, col = "blue")
  
  res.pip = compare_pip(pip.susie, pip.rss, is_effect)
  plot_pip(res.pip, labs = c("susie", "rss"), main = paste0("censor=",censor_lvls[i]))
  abline(a = 0, b = 1, lty = 2, col = "blue")
  
}
```

```{r fig.width= 12, fig.height=18}
par(mfrow = c(5,5))
for (i in 1:length(censor_lvls)){
  indx = which(bvsnlp$simulate.cor_type == "independent" & bvsnlp$simulate.censor_lvl == censor_lvls[i])
  is_effect = unlist(lapply(indx, function(x) bvsnlp$simulate.is_effect[[x]]))
  pip.susie = unlist(lapply(indx, function(x) susie$susie.pip[[x]]))
  pip.survsvb = unlist(lapply(indx, function(x) svb$svb.pip[[x]]))
  pip.bvsnlp = unlist(lapply(indx, function(x) bvsnlp$bvsnlp.pip[[x]]))
  pip.rss = unlist(lapply(indx, function(x) rss$rss.pip[[x]]))
  pip.r2b = unlist(lapply(indx, function(x) r2b$r2b.pip[[x]]))
  
  # Calibration plot for pip.susie
  calibration.susie = pip_calibration(pip.susie, is_effect)
  plot_calibration(calibration.susie, main = paste0("susie: censor=", censor_lvls[i]))
  
  # Calibration plot for pip.svb
  calibration.survsvb = pip_calibration(pip.survsvb, is_effect)
  plot_calibration(calibration.survsvb, main = paste0("svb: censor=", censor_lvls[i]))
  
  # Calibration plot for pip.bvsnlp
  calibration.bvsnlp = pip_calibration(pip.bvsnlp, is_effect)
  plot_calibration(calibration.bvsnlp, main = paste0("bvsnlp: censor=", censor_lvls[i]))
  
  # Calibration plot for pip.rss
  calibration.rss = pip_calibration(pip.rss, is_effect)
  plot_calibration(calibration.rss, main = paste0("rss: censor=", censor_lvls[i]))
  
  # Calibration plot for pip.r2b
  calibration.r2b = pip_calibration(pip.r2b, is_effect)
  plot_calibration(calibration.r2b, main = paste0("r2b: censor=", censor_lvls[i]))
}
```
