---
title: Introduction to CoxPH-SuSiE
author: Yunqi Yang, Karl Tayeb, Peter Carbonetto and Matthew Stephens
date: December 10, 2024
output:
  workflowr::wflow_html:
    theme: readable
    toc: no
    highlight: textmate
    lib_dir: site_libs
    self_contained: no
---

This short vignette is intended to (i) introduce CoxPH-SuSiE for
fine-mapping time-to-event (TTE) phenotypes, and (ii) illustrate how
to use CoxPH-SuSiE on an example genetic data set.

```{r knitr-opts, include=FALSE}
knitr::opts_chunk$set(comment = "#",collapse = TRUE,results = "hold",
                      fig.align = "center",dpi = 120)
```

## Initial setup

To run the code below, you will need the survival, susieR and
logisticsusie R packages, which can be installed with these commands
in R:

```{r install-packages, eval=FALSE}
install.packages("remotes")
install.packages("survival")
remotes::install_github("stephenslab/susieR")
remotes::install_github("karltayeb/logisticsusie")
```

Also, we will use the ggplot2 and cowplot packages to create some
helpful visualizations of the data and results:

```{r install-more-packages, eval=FALSE}
install.packages("ggplot2")
install.packages("cowplot")
```

Once the packages are installed, load them.

```{r load-packages, message=FALSE}
library(survival)
library(susieR)
library(logisticsusie)
library(ggplot2)
library(cowplot)
```

Set the seed for reproducibility.

```{r set-seed}
set.seed(1)
```

## The data set

The example data set is found in the "data" directory of this GitHub
repository, or can be downloaded by running these commands in R:

```{r load-data}
data_url <- paste0("https://raw.githubusercontent.com/",
                   "yunqiyang0215/survival-susie/",
			       "refs/heads/master/data/survival_demo.RData")
download.file(data_url,"survival_demo.RData")
load("survival_demo.RData")
```

In this example, the TTE data are "censored" disease diagnosis times,
meaning that the observations are either: the time of diagnosis (for
uncensored samples); or a lower limit on the time of diagnosis for
censored samples. The censored samples are indicated by the "+" (this
is called "right censoring"):

```{r survival-data}
pheno <- Surv(time,status)
head(pheno,n = 10)
```

Among the 574 diagnosis times in this data set, 330 are observed, and
the remaining 244 are right-censored:

```{r survival-data-2}
table(status)
```

The "geno" matrix contains the (imputed) genotype data for 1,000
candidate genetic variants (SNPs) located near each other on the
chromosome.

```{r geno-data}
nrow(geno)
ncol(geno)
```

## Association analysis using the CoxPH model

Association tests performed using the Cox proportional hazards (CoxPH)
model show that many of thee genetic variants are associated with
diagnosis time:

```{r genome-wide-scan, fig.height=2, fig.width=6}
pvals <-
  apply(geno,2,
        function (x) summary(coxph(pheno ~ x))$coefficients[1,"Pr(>|z|)"])
pdat <- data.frame(pos = 1:1000,pval = pvals,causal = (b != 0))
ggplot(pdat,aes(x = pos,y = -log10(pval),color = causal,shape = causal)) +
  geom_point(size = 1.25) +
  scale_color_manual(values = c("black","darkorange")) +
  scale_shape_manual(values = c(1,17)) +
  labs(x = "chromosomal position",y = "-log10 p-value") +
  theme_cowplot(font_size = 10)
```

Since we simulated these data, we know that 3 of the 1,000 SNPs truly
affect the TTE phenotype (drawn in orange in the plot above). All
three of these "causal" SNPs show evidence for association from the
CoxPH-based association tests, but the strength of support for
association varies quite a bit among the three SNPs. Further, many
more SNPs that do not affect the phenotype are also strongly
associated with the phenotype. This happens *because many of the SNPs
are strongly correlated with each other* (in strong *linkage
disequilibrium*, or "LD").

Next we will run the CoxPH-SuSiE fine-mapping method to try and
narrow the association signal down to a smaller set of causal
candidates.

## Fine-mapping analysis using the CoxPH model

To run the fine-mapping, we first need to define a function that
performs a (Bayesian) univariate association analysis (here, using the
CoxPH model). More precisely, this function takes as input the
regression data (X, Y, as well as an additional "offset" term), and
the prior variance of the regression coefficient (v0), and outputs
four quantities: (1) the posterior mean of the regression coefficient;
(2) its posterior variance; (3) the log-Bayes factor comparing the
model against the "null" model (when the coefficient is zero); and (4)
an estimate of the prior variance parameter (here we use the
maximum-likelihood estimate, which is simply the posterior second
moment of the regression coefficient. (This last quantity can be a
constant if you want to keep the prior variance at a fixed value.)

```{r surv-uni-fun}
surv_uni_fun <- function (x, y, e, prior_variance,
                          estimate_intercept = 0, ...) {
  v0   <- prior_variance						 
  fit  <- coxph(y ~ x + offset(e))
  out  <- summary(fit)$coefficients
  bhat <- out[1,"coef"]
  s    <- out[1,"se(coef)"]
  z    <- bhat/s
  lbf  <- log(s^2/(v0 + s^2))/2 + z^2/2*v0/(v0 + s^2)
  lbf  <- lbf - z^2/2 + summary(fit)$logtest["test"]/2
  v1   <- 1/(1/v0 + 1/s^2)
  mu1  <- v1*bhat/s^2
  return(list(mu = mu1,var = v1,lbf = lbf,
              prior_variance = mu1^2 + v1,
			  intercept = 0))
}
```

Now we can fit the CoxPH-based SuSiE model using the `ibss_from_ser()`
function from the logisticsusie package. Note that this step may
take a minute or two to run depending on the computing platform:

```{r ibss-from-ser, warning=FALSE}
t0 <- proc.time()
fit <- ibss_from_ser(geno,pheno,L = 3,tol = 0.0001,maxit = 100,
                     ser_function = ser_from_univariate(surv_uni_fun))
t1 <- proc.time()
print(t1 - t0)
```

(As an aside, note that the univariate regression computations are
trivially parallelizable, so you have a multicore computing platform,
you can make the method run faster, e.g.,

```{r ibss-from-ser-parallel, warning=FALSE}
t0 <- proc.time()
fit <- ibss_from_ser(geno,pheno,L = 3,tol = 0.0001,maxit = 100,
                     ser_function = ser_from_univariate(surv_uni_fun),
					 num_cores = 4)
t1 <- proc.time()
print(t1 - t0)
```

In this run we were able to achieve about a 3x speedup with 4 threads.)

Now we use the `susie_get_cs()` function from the susieR package to
compute credible sets (CSs):

```{r credible-sets}
class(fit) <- c("susie","list")
out <- susie_get_cs(fit,geno,coverage = 0.95,min_abs_cor = 0.1)
out$cs
```

Interestingly, in one credible set, CoxPH-SuSiE is able to identify a
single causal SNP with very high confidence (shown in blue in the
plot). Additionally, a second set of causal candidates is identified
with 6 plausible candidates (shown in orange). *Both of these CSs
contain the true causal SNPs.*

```{r plot-credible-sets, fig.height=2, fig.width=6}
pdat <- cbind(pdat,CS = 0)
pdat[out$cs$L1,"CS"] <- 1
pdat[out$cs$L2,"CS"] <- 2
pdat <- transform(pdat,CS = factor(CS))
cs_colors <- c("black","darkorange","dodgerblue")
ggplot(pdat,aes(x = pos,y = -log10(pval),color = CS,shape = causal)) +
  geom_point(size = 1.25) +
  scale_color_manual(values = cs_colors) +
  scale_shape_manual(values = c(1,17)) +
  labs(x = "chromosomal position",y = "-log10 p-value") +
  theme_cowplot(font_size = 10)
```

The PIPs ("posterior inclusion probabilities") measure the relative
support for the SNPs within each CS. Plotting the PIPs for all SNPs is
helpful for understanding precisely which SNPs are the top candidates:

```{r pip-plot, fig.height=2, fig.width=6}
pdat <- cbind(pdat,pip = fit$pip)
ggplot(pdat,aes(x = pos,y = pip,color = CS,shape = causal)) +
  geom_point(size = 1.25) +
  scale_color_manual(values = cs_colors) +
  scale_shape_manual(values = c(1,17)) +
  labs(x = "chromosomal position",y = "PIP") +
  theme_cowplot(font_size = 10)
```

For example, in CS 1 (orange), 3 out of the 6 SNPs are much stronger
candidates than the others.

It is also interesting that the third causal SNPs that was not
identified does show some modest evidence for being a causal SNP in
the PIP plot.

Let's compare the coefficient estimates against the ground-truth
estimates. These are contained in the "mu" output. Here we see that
the first CS (L1) accurately estimates the coefficient for the second
causal SNP (the true value is -2), and the second CS (L2) accurately
estimates the coefficient for the third causal SNP (the true value is
2).

```{r compare-coefs}
rownames(fit$mu) <- c("L1","L2","L3")
causal_snps <- which(b != 0)
round(rbind(b,fit$mu)[,causal_snps],digits = 3)
```

Finally, note that CoxPH-SuSiE is robust to overestimating the number
of causal SNPs. For example, suppose we set the number of causal SNPs
to 10:

```{r ibss-from-ser-L-10, warning=FALSE, results="hide"}
fit <- ibss_from_ser(geno,pheno,L = 10,tol = 0.0001,maxit = 100,
                     ser_function = ser_from_univariate(surv_uni_fun))
```

The log-Bayes factors quantifying support for each CS (i.e., each
causal SNP) are in the "lbf_ser" output:

```{r ibss-bf}
fit$lbf_ser
```

After the first two CSs, the remaining 8 CSs provide very little
improvement in fit. This result tells us that we can focus on the
results for the first two CSs and ignore the rest. (Note these
log-Bayes factors are computed using the natural logarithm.)
