---
title: Introduction to CoxPH-SuSiE
author: Yunqi Yang and Peter Carbonetto
date: December 6, 2024
output:
  workflowr::wflow_html:
    theme: readable
    toc: no
    highlight: textmate
    lib_dir: site_libs
    self_contained: no
---

This short vignette is intended to introduce the CoxPH-SuSiE model for
fine-mapping time-to-event (TTE) phenotypes, and illustrate how to use
the methods on an example genetic data set.

```{r knitr-opts, include=FALSE}
knitr::opts_chunk$set(comment = "#",collapse = TRUE,results = "hold",
                      fig.align = "center",dpi = 120)
```

## Initial setup

To run the code below, you will need the survival, susieR and
logisticsusie R packages, which can be installed with these commands
in R:

```{r install-packages, eval=FALSE}
install.packages("remotes")
install.packages("survival")
remotes::install_github("stephenslab/susieR")
remotes::install_github("karltayeb/logisticsusie")
```

Also, we will use the ggplot2 and cowplot packages to create some
helpful visualizations of the data and results:

```{r install-more-packages, eval=FALSE}
install.packages(c("ggplot2","cowplot"))
```

Once the packages are installed, load them.

```{r load-packages, message=FALSE}
library(survival)
library(susieR)
library(logisticsusie)
library(ggplot2)
library(cowplot)
```

## The data set

The example data set is found in the "data" directory of this
repository, or can be downloaded by running these commands in R:

```{r load-data}
data_url <- paste0("https://raw.githubusercontent.com/",
                   "yunqiyang0215/survival-susie/",
			       "refs/heads/master/data/survival_demo.RData")
download.file(data_url,"survival_demo.RData")
load("survival_demo.RData")
```

In this example, the TTE data are "censored" disease diagnosis times,
meaning that the observations are either: the time of diagnosis for
uncensored samples; or a lower limit on the time of diagnosis for
censored samples, indicated by the "+". (This is called "right
censoring".)

```{r survival-data}
pheno <- Surv(time,status)
head(pheno,n = 10)
```

Among the 574 diagnosis times in this data set, 330 are observed, and
the remaining 244 are right-censored:

```{r survival-data-2}
table(status)
```

The "geno" matrix contains the (imputed) genotype data for 1,000
candidate genetic variants (SNPs) positioned near each other on a
chromosome.

```{r geno-data}
nrow(geno)
ncol(geno)
```

## Association analysis using the CoxPH model

Basic association tests using the Cox proportional hazards (CoxPH)
model show that many of these genetic variants are associated with
the TTE phenotype:

```{r genome-wide-scan, fig.height=2, fig.width=6}
pvals <-
  apply(geno,2,
        function (x) summary(coxph(pheno ~ x))$coefficients[1,"Pr(>|z|)"])
pdat <- data.frame(pos = 1:1000,pval = pvals,causal = (b != 0))
ggplot(pdat,aes(x = pos,y = -log10(pval),color = causal,shape = causal)) +
  geom_point(size = 1.25) +
  scale_color_manual(values = c("dodgerblue","darkorange")) +
  scale_shape_manual(values = c(1,17)) +
  labs(x = "chromosomal position",y = "-log10 p-value") +
  theme_cowplot(font_size = 10)
```

Since we simulated these data, we know that three SNPs truly affect
the TTE phenotype (drawn in orange in the plot). All three of the
causal SNPs show evidence for association from the CoxPH-based
association tests, but the strength of support for association varies
quite a bit among the three SNPs. Further, many more SNPs that do not
affect the phenotype are also associated, and this occurs *because
many of the SNPs are strongly correlated with each other* (strong
*linkage disequilibrium*, or "LD" for short).

Next we will run the CoxPH-SuSiE fine-mapping method in an attempt to
narrow the association signal down to a smaller set of causal
candidates.

## Fine-mapping analysis using the CoxPH model

```{r surv-uni-fun}
surv_uni_fun <- function (x, y, e, v0, estimate_intercept = 0, ...) {
  fit  <- coxph(y ~ x + offset(e))
  out  <- summary(fit)$coefficients
  bhat <- out[1,"coef"]
  s    <- out[1,"se(coef)"]
  z    <- bhat/s
  lbf  <- log(s^2/(v0 + s^2))/2 + z^2/2*v0/(v0 + s^2)
  lbf  <- lbf - z^2/2 + summary(fit)$logtest["test"]/2
  v1   <- 1/(1/v0 + 1/s^2)
  mu1  <- v1*bhat/s^2
  return(list(mu = mu1,var = v1,lbf = lbf,intercept = 0))
}
```

This step may take a minutes or two to run.

```{r ibss-from-ser, warning=FALSE, results="hide"}
fit <- ibss_from_ser(geno,pheno,L = 3,tol = 0.0001,maxit = 100,
                     ser_function = ser_from_univariate(surv_uni_fun))
```

```{r credible-sets}
class(fit) <- c("susie","list")
out <- susie_get_cs(fit,geno,min_abs_cor = 0.1)
out$cs
```

```{r plot-credible-sets, fig.height=2, fig.width=6}
pdat <- cbind(pdat,CS = 0)
pdat[out$cs$L1,"CS"] <- 1
pdat[out$cs$L2,"CS"] <- 2
pdat <- transform(pdat,CS = factor(CS))
cs_colors <- c("black","darkorange","dodgerblue")
ggplot(pdat,aes(x = pos,y = -log10(pval),color = CS,shape = causal)) +
  geom_point(size = 1.25) +
  scale_color_manual(values = cs_colors) +
  scale_shape_manual(values = c(1,17)) +
  labs(x = "chromosomal position",y = "-log10 p-value") +
  theme_cowplot(font_size = 10)
```

```{r pip-plot, fig.height=2, fig.width=6}
pdat <- cbind(pdat,pip = fit$pip)
ggplot(pdat,aes(x = pos,y = pip,color = CS,shape = causal)) +
  geom_point(size = 1.25) +
  scale_color_manual(values = cs_colors) +
  scale_shape_manual(values = c(1,17)) +
  labs(x = "chromosomal position",y = "PIP") +
  theme_cowplot(font_size = 10)
```
