---
title: "SPAcox"
author: "Yunqi Yang"
date: "9/11/2023"
output: html_document
---

### 1. Understand SPAcox

Result format: an R matrix with the following columns

1. MAF: Minor allele frequencies

2. missing.rate: Missing rates

3. p.value.spa: p value (recommanded) from a saddlepoint approximation.

4. p.value.norm: p value from a normal distribution approximation.

5. Stat: score statistics

6. Var: estimated variances of the score statistics

7. z: z values corresponding to the score statistics

```{r}
library(SPACox)
library(survival)
```

```{r}
# Simulation phenotype and genotype
set.seed(1)
N = 1e3
nSNP = 3
MAF = 0.1
Phen.mtx = data.frame(ID = paste0("IID-",1:N),
                      event=rbinom(N,1,0.5),
                      time=runif(N),
                      Cov1=rnorm(N),
                      Cov2=rbinom(N,1,0.5))

# First two SNPs with cor = 1
x1 = rnorm(N)
x2 = x1 + rnorm(N, sd = 1)
Geno.mtx = matrix(c(x1, x1, x2),N,nSNP)
cor(Geno.mtx)

# NOTE: The row and column names of genotype matrix are required.
rownames(Geno.mtx) = paste0("IID-",1:N)
colnames(Geno.mtx) = paste0("SNP-",1:nSNP)
```

```{r}
# Attach the survival package so that we can use its function Surv()
t1 = proc.time()
obj.null = SPACox_Null_Model(Surv(time,event)~Cov1+Cov2, data=Phen.mtx,
                             pIDs=Phen.mtx$ID, gIDs=rownames(Geno.mtx))
SPACox.res = SPACox(obj.null, Geno.mtx)
t2 = proc.time()

t2 - t1


# The below is an example code to use survival package
t1 = proc.time()
coxph(Surv(time,event)~Cov1+Cov2+Geno.mtx, data=Phen.mtx)
t2 = proc.time()

t2 - t1
```

```{r}
# we recommand using column of 'p.value.spa' to associate genotype with time-to-event phenotypes
head(SPACox.res)
```

```{r}
coxph(Surv(time,event)~Cov1+Cov2+Geno.mtx[,1], data=Phen.mtx)
coxph(Surv(time,event)~Cov1+Cov2+Geno.mtx[,3], data=Phen.mtx)
```


### 2.Speed comparison between SPAcox and coxph()

```{r}
# Simulation phenotype and genotype
set.seed(1)
N = 10000
nSNP = 1000
MAF = 0.1
Phen.mtx = data.frame(ID = paste0("IID-",1:N),
                      event=rbinom(N,1,0.5),
                      time=runif(N),
                      Cov1=rnorm(N),
                      Cov2=rbinom(N,1,0.5))
Geno.mtx = matrix(rbinom(N*nSNP,2,MAF),N,nSNP)

# NOTE: The row and column names of genotype matrix are required.
rownames(Geno.mtx) = paste0("IID-",1:N)
colnames(Geno.mtx) = paste0("SNP-",1:nSNP)


t1 = proc.time()
obj.null = SPACox_Null_Model(Surv(time,event)~Cov1+Cov2, data=Phen.mtx,
                             pIDs=Phen.mtx$ID, gIDs=rownames(Geno.mtx))
SPACox.res = SPACox(obj.null, Geno.mtx)
t2 = proc.time()
time.spa = t2 - t1


# The below is an example code to use survival package
t1 = proc.time()
coxph(Surv(time,event)~Cov1+Cov2+Geno.mtx[,1], data=Phen.mtx)
t2 = proc.time()
time.coxph = (t2 - t1)*nSNP

```

```{r}
time.spa
time.coxph
```

